### PR TITLE
Feature/paginated fetching scheduled events

### PR DESCRIPTION
**Pull request by davidebukali**
Fix for #206

### CODE REVIEW

**Review by edwinmp — CHANGES_REQUESTED** [review]

**Review by edwinmp — COMMENTED** [review]
Could you please fetch them sorted by descending start date (as they were before).

Also add loading indicators for when data is being fetched. There are already tables that have this, so customise from those.

File: frontend/src/components/ScheduledEventsRunHistoryTableCard/ScheduledEventsRunHistoryTableCard.tsx [hunk 1]
```diff
@@ -20,13 +20,14 @@ export const ScheduledEventsRunHistoryTableCard: FunctionComponent<ComponentProp
...
     if (event) {
-      fetchScheduledEventRunHistory(event.id).then((result) => {
-        setHistoryData(result.data);
-        setCount(result.data.length);
-        setPageCount(Math.ceil(result.data.length / LIMIT));
+      fetchDataPerPage(event.id, LIMIT, currentPage).then((result) => {
+        setHistoryData(result.data.results);
+        setCount(result.data.count);
+        setPageCount(Math.ceil(result.data.count / LIMIT));
+        console.log(`Page count is ${pageCount}`);
```
[the remark this answers is not in the document]
Comment (davidebukali) [reply frontend/src/components/ScheduledEventsRunHistoryTableCard/ScheduledEventsRunHistoryTableCard.tsx]: Yes, removed ...

File: frontend/src/components/ScheduledEventsTable/ScheduledEventsTable.tsx [hunk 2]
```diff
@@ -13,12 +13,20 @@ export interface ScheduledEventTableProps {
 export const ScheduledEventsTable: FunctionComponent<ScheduledEventTableProps> = (props) => {
   const { activeEvent } = useContext(ScheduledEventContext);
 
+  const getRowNumber = (index: number): number => {
+    if (props.currentPage === 1) {
+      return index + 1;
+    } else {
+      return (props.currentPage - 1) * props.pageLimit + 1;
+    }
```
[the remark this answers is not in the document]
Comment (davidebukali) [reply frontend/src/components/ScheduledEventsTable/ScheduledEventsTable.tsx]: Updated

**Review by edwinmp — APPROVED** [review]
LGTM!

**Pull request merged**